app/controllers/bots_controller.rb

```diff
@@ -2,59 +2,59 @@ class BotsController < ApplicationController
   before_action :set_bot, only: [:show, :edit, :update, :destroy]
   before_action :authenticate_user!, except: [:index, :show]
   before_action :check_bot_ownership, only: [:edit, :update, :destroy]
-
+  
```

Comment:
There are a few of these weird little empty two-space blocks floating around, would be good to remove them.